syne_tune/optimizer/schedulers/multiobjective/multi_objective_searcher.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+import logging
+from typing import Optional, List
+
+from syne_tune.optimizer.schedulers.searchers.searcher import (
+    BaseSearcher,
+    impute_points_to_evaluate,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+class BaseMultiObjectiveSearcher(BaseSearcher):
+    """
+    Base class for searchers that optimize multiple objectives simultaneously.
+    Derivatives of this class need to implement :meth:`~get_config`.
+
+    :param config_space: Configuration space
+    :param metrics: Name of all metrics that should be optimized.
+    :param points_to_evaluate: List of configurations to be evaluated
+        initially (in that order). Each config in the list can be partially
+        specified, or even be an empty dict. For each hyperparameter not
+        specified, the default value is determined using a midpoint heuristic.
+        If ``None`` (default), this is mapped to ``[dict()]``, a single default config
+        determined by the midpoint heuristic. If ``[]`` (empty list), no initial
+        configurations are specified.
+    :param modes: Defines for each metric if it should be minimized ("min", default) or maximized
+        ("max")
+    """
+
+    def __init__(
+        self,
+        config_space: dict,
+        metrics: List[str],
+        modes: Optional[List[dict], str] = "min",
+        points_to_evaluate: Optional[List[dict]] = None,
+    ):
+        self.config_space = config_space
+        self._metrics = metrics
+        self._points_to_evaluate = impute_points_to_evaluate(
```

Comment:
I'd still rather call the superclass constructor. You can then set `_metric` and `_mode` to None if you like.

Reply:
What is the benefit here of calling the superclass constructor? It might be confusing for developers if we pass None to `_metric` and `_mode` here.